sycl/include/CL/sycl/detail/pi.h

```diff
@@ -1648,6 +1654,19 @@ __SYCL_EXPORT pi_result piextPluginGetOpaqueData(void *opaque_data_param,
 /// \param PluginParameter placeholder for future use, currenly not used.
 __SYCL_EXPORT pi_result piTearDown(void *PluginParameter);
 
+/// API to get extension name for those extension whose name varies with backend
+/// \param ExtNumber number of extension \sa pi_extension_number
+/// \param Size[out] length of returned extension name
+/// \param Value[out] memory location where to put extension name
+/// \return \c PI_SUCCESS if this extension name is available,
+///         \c PI_INVALID_VALUE if the extension name isn't available for this
+///         plugin
+///
+/// Either of Size or Value isn't set if it's nil.
+/// Value returned in Size can't be zero.
+__SYCL_EXPORT pi_result piextGetExtensionName(pi_extension_number ExtNumber,
```

Comment:
Could you please clarify why this API is needed?
FYI @smaslov-intel

Reply:
As was stated in design review, name of extension can vary with backend. In order to allow this behaviour the new API was introduced.

Reply:
How is that name to be used in SYCL RT? Don't you just need to known if "assert" (or other extension) is supported by the backend, and thus can just return bool?

Reply:
I think we should re-use piPlatformGetInfo/PI_PLATFORM_INFO_EXTENSIONS and define a new extension name e.g. "cassert", which will be returned by every backend/platform supporting asserts.